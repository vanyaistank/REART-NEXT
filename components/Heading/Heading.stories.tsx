// import React from 'react';
//
// import { storiesOf } from '@storybook/react';
//
// import { Heading } from '@Components';
//
// storiesOf('Heading', module)
// 	.add('default prop XL', () => (
// 		<Heading mono darkGray>
// 			HEADING
// 		</Heading>
// 	))
// 	.add('L', () => <Heading size="L">HEADING</Heading>)
// 	.add('M', () => <Heading size="M">HEADING</Heading>)
// 	.add('S', () => <Heading size="S">HEADING</Heading>)
// 	.add('mono dark gray Heading', () => (
// 		<Heading mono darkGray>
// 			HEADING
// 		</Heading>
// 	))
// 	.add('bold and uppercase lighter gray Heading', () => (
// 		<Heading bold uppercase lighterGray>
// 			HEADING
// 		</Heading>
// 	))
// 	.add('black (default color prop) Heading with vertical margins', () => (
// 		<Heading marginBottom="30px" marginTop="30px">
// 			HEADING
// 		</Heading>
// 	))
// 	.add('purple Heading', () => <Heading purple>HEADING</Heading>)
// 	.add('dark purple Heading', () => <Heading darkPurple>HEADING</Heading>);
