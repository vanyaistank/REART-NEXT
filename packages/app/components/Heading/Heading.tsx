import React from 'react';
import styledProps from 'styled-props';

import styled, { css } from 'styled-components';
import {
	fontSize,
	colors,
	colorsPartialInterface,
	headingMargin,
} from '@Styled/theme';

interface Props extends colorsPartialInterface {
	size?: 'XL' | 'L' | 'M' | 'S';
	mono?: boolean;
	inline?: boolean;
	bold?: boolean;
	children: React.ReactNode;
	className?: string;
	uppercase?: boolean;
	marginTop?: string;
	marginBottom?: string;
	onClick?: () => void;
	centered?: boolean;
	color?: string; // default value for styledProps
}

const Sizes = {
	XL: 1,
	L: 2,
	M: 3,
	S: 4,
};

const Heading: React.SFC<Props> = ({
	size = 'XL',
	className,
	onClick = () => {},
	children,
}) => React.createElement(`h${Sizes[size]}`, { className, onClick }, children);

const StyledHeading = styled(Heading)`
	${(props: Props) => {
		const handleSize = (param): string => param || headingMargin[props.size];
		return css`
			display: ${props.inline && 'inline'};
			font-size: ${fontSize[props.size]};
			font-family: ${props.mono
		? 'Source Code Pro, monospace'
		: 'Lato, sans-serif'};
			font-weight: ${props.bold && 'bold'};
			color: ${styledProps(colors, 'color')};
			text-transform: ${props.uppercase && 'uppercase'};
			margin-top: ${handleSize(props.marginTop)};
			margin-bottom: ${handleSize(props.marginBottom)};
			text-align: ${props.centered && 'center'};
		`;
	}};
`;

StyledHeading.defaultProps = {
	size: 'XL',
	mono: false,
	bold: false,
	uppercase: false,
	marginTop: null,
	marginBottom: null,
	color: 'white',
};

export default StyledHeading;
