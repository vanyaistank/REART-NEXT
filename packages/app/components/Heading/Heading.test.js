import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import sinon from 'sinon';
import Heading from './Heading';

const sizes = ['XL', 'L', 'M', 'S'];

for (let i = 0; i < sizes.length; i++) {
	const currentSize = sizes[i];

	it(`renders black ${currentSize} Heading`, () => {
		const wrapper = shallow(
			<Heading black size={currentSize}>
				$
				{currentSize}
			</Heading>
		);
		expect(wrapper).toMatchSnapshot();
	});
}

it('simulates click events', () => {
	const onHeadingClick = sinon.spy();
	const wrapper = shallow(<Heading onClick={onHeadingClick}>CLICK!</Heading>);
	wrapper.find('Heading').simulate('click');
	expect(onHeadingClick.calledOnce).toEqual(true);
});
