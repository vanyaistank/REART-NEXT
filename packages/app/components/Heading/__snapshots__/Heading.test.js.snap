// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders black L Heading 1`] = `
.c0 {
  font-size: 30px;
  font-family: Lato,sans-serif;
  color: #000000;
  margin-top: 20px;
  margin-bottom: 20px;
}

<Heading
  black={true}
  bold={false}
  className="c0"
  marginBottom={null}
  marginTop={null}
  mono={false}
  size="L"
  uppercase={false}
>
  $
  L
</Heading>
`;

exports[`renders black M Heading 1`] = `
.c0 {
  font-size: 17px;
  font-family: Lato,sans-serif;
  color: #000000;
  margin-top: 15px;
  margin-bottom: 15px;
}

<Heading
  black={true}
  bold={false}
  className="c0"
  marginBottom={null}
  marginTop={null}
  mono={false}
  size="M"
  uppercase={false}
>
  $
  M
</Heading>
`;

exports[`renders black S Heading 1`] = `
.c0 {
  font-size: 15px;
  font-family: Lato,sans-serif;
  color: #000000;
  margin-top: 10px;
  margin-bottom: 10px;
}

<Heading
  black={true}
  bold={false}
  className="c0"
  marginBottom={null}
  marginTop={null}
  mono={false}
  size="S"
  uppercase={false}
>
  $
  S
</Heading>
`;

exports[`renders black XL Heading 1`] = `
.c0 {
  font-size: 40px;
  font-family: Lato,sans-serif;
  color: #000000;
  margin-top: 50px;
  margin-bottom: 50px;
}

<Heading
  black={true}
  bold={false}
  className="c0"
  marginBottom={null}
  marginTop={null}
  mono={false}
  size="XL"
  uppercase={false}
>
  $
  XL
</Heading>
`;
